src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsRulesSpec.groovy

```diff
@@ -988,4 +1000,139 @@ class PriceFloorsRulesSpec extends PriceFloorsBaseSpec {
+        }
+        floorsProvider.setResponse(bidRequest.accountId, floorsResponse)
+
+        and: "PBS fetch rules from floors provider"
+        cacheFloorsProviderRules(floorsPbsService, bidRequest)
+
+        when: "PBS processes auction request"
+        def response = floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request bidFloor should correspond to appropriate rule"
+        def bidderRequest = getRequest(response)
+        assert bidderRequest.size() == 2
+        assert bidderRequest[GENERIC.value].first.imp[0].bidFloor == genericBidFloorRuleValue
+        assert bidderRequest[OPENX.value].first.imp[0].bidFloor == openxBidFloorRuleValue
+
+        and: "Bidder request should contain proper bid floor value"
+        def bidderRequests = bidder.getBidderRequests(bidRequest.id)
+        assert bidderRequests.imp.bidFloor.toSet().flatten()
+                .any { bidFloor -> bidFloor == genericBidFloorRuleValue || bidFloor == openxBidFloorRuleValue }
+
+        where:
+        genericBidFloorRuleValue  | openxBidFloorRuleValue
+        null                      | PBSUtils.randomFloorValue
+        PBSUtils.randomFloorValue | null
+        PBSUtils.randomFloorValue | PBSUtils.randomFloorValue
+    }
+
+    def "PBS shouldn't apply rule for alias bidder when rule contain main bidder"() {
+        given: "BidRequest with generic and alias bidder"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.aliases = [(ALIAS.value): GENERIC]
+            imp[0].ext.prebid.bidder.alias = new Generic()
+        }
+
+        and: "Account with enabled fetch, fetch.url in the DB"
+        def account = getAccountWithEnabledFetch(bidRequest.accountId)
+        accountDao.save(account)
+
+        and: "Set Floors Provider response"
+        def floorsResponse = PriceFloorData.priceFloorData.tap {
+            modelGroups[0].schema = new PriceFloorSchema(fields: [BIDDER])
+            modelGroups[0].values = [(new Rule(bidder: GENERIC).rule)  : genericBidFloorRuleValue]
+        }
+        floorsProvider.setResponse(bidRequest.accountId, floorsResponse)
+
+        and: "PBS fetch rules from floors provider"
+        cacheFloorsProviderRules(floorsPbsService, bidRequest)
+
+        when: "PBS processes auction request"
+        def response = floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request bidFloor should correspond to appropriate rule"
+        def bidderRequest = getRequest(response)
+        assert bidderRequest.size() == 2
+        assert bidderRequest[GENERIC.value].first.imp[0].bidFloor == genericBidFloorRuleValue
+        assert !bidderRequest[ALIAS.value].first.imp[0].bidFloor
+
+        and: "Bidder request should contain proper bid floor value"
+        def bidderRequests = bidder.getBidderRequests(bidRequest.id)
+        assert bidderRequests.imp.bidFloor.toSet().flatten()
+                .any { bidFloor -> bidFloor == genericBidFloorRuleValue }
```

Comment:
Again, bad check. Test reads that floor for main material bidder shouldn't influence alias bidder, but you check that ANY floor value should correspond to a main bidder.